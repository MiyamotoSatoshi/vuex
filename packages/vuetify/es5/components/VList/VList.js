'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; // Styles

// Mixins


require('../../../src/stylus/components/_lists.styl');

var _themeable = require('../../mixins/themeable');

var _themeable2 = _interopRequireDefault(_themeable);

var _registrable = require('../../mixins/registrable');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* @vue/component */
exports.default = {
    name: 'v-list',
    mixins: [(0, _registrable.provide)('list'), _themeable2.default],
    provide: function provide() {
        return {
            'listClick': this.listClick
        };
    },

    props: {
        dense: Boolean,
        expand: Boolean,
        subheader: Boolean,
        threeLine: Boolean,
        twoLine: Boolean
    },
    data: function data() {
        return {
            groups: []
        };
    },
    computed: {
        classes: function classes() {
            return _extends({
                'v-list--dense': this.dense,
                'v-list--subheader': this.subheader,
                'v-list--two-line': this.twoLine,
                'v-list--three-line': this.threeLine
            }, this.themeClasses);
        }
    },
    methods: {
        register: function register(uid, cb) {
            this.groups.push({ uid: uid, cb: cb });
        },
        unregister: function unregister(uid) {
            var index = this.groups.findIndex(function (g) {
                return g.uid === uid;
            });
            if (index > -1) {
                this.groups.splice(index, 1);
            }
        },
        listClick: function listClick(uid) {
            if (this.expand) return;
            for (var i = this.groups.length; i--;) {
                this.groups[i].cb(uid);
            }
        }
    },
    render: function render(h) {
        var data = {
            staticClass: 'v-list',
            'class': this.classes
        };
        return h('div', data, [this.$slots.default]);
    }
};
//# sourceMappingURL=VList.js.map