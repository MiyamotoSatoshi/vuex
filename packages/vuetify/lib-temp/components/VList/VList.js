// Styles
import '../../stylus/components/_lists.styl';
// Mixins
import Themeable from '../../mixins/themeable';
import { provide as RegistrableProvide } from '../../mixins/registrable';
/* @vue/component */
export default {
    name: 'v-list',
    mixins: [
        RegistrableProvide('list'),
        Themeable
    ],
    provide() {
        return {
            'listClick': this.listClick
        };
    },
    props: {
        dense: Boolean,
        expand: Boolean,
        subheader: Boolean,
        threeLine: Boolean,
        twoLine: Boolean
    },
    data: () => ({
        groups: []
    }),
    computed: {
        classes() {
            return {
                'v-list--dense': this.dense,
                'v-list--subheader': this.subheader,
                'v-list--two-line': this.twoLine,
                'v-list--three-line': this.threeLine,
                ...this.themeClasses
            };
        }
    },
    methods: {
        register(uid, cb) {
            this.groups.push({ uid, cb });
        },
        unregister(uid) {
            const index = this.groups.findIndex(g => g.uid === uid);
            if (index > -1) {
                this.groups.splice(index, 1);
            }
        },
        listClick(uid) {
            if (this.expand)
                return;
            for (let i = this.groups.length; i--;) {
                this.groups[i].cb(uid);
            }
        }
    },
    render(h) {
        const data = {
            staticClass: 'v-list',
            'class': this.classes
        };
        return h('div', data, [this.$slots.default]);
    }
};
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiVkxpc3QuanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi9zcmMvY29tcG9uZW50cy9WTGlzdC9WTGlzdC5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQSxTQUFTO0FBQ1QsT0FBTyxxQ0FBcUMsQ0FBQTtBQUU1QyxTQUFTO0FBQ1QsT0FBTyxTQUFTLE1BQU0sd0JBQXdCLENBQUE7QUFDOUMsT0FBTyxFQUNMLE9BQU8sSUFBSSxrQkFBa0IsRUFDOUIsTUFBTSwwQkFBMEIsQ0FBQTtBQUVqQyxvQkFBb0I7QUFDcEIsZUFBZTtJQUNiLElBQUksRUFBRSxRQUFRO0lBRWQsTUFBTSxFQUFFO1FBQ04sa0JBQWtCLENBQUMsTUFBTSxDQUFDO1FBQzFCLFNBQVM7S0FDVjtJQUVELE9BQU87UUFDTCxPQUFPO1lBQ0wsV0FBVyxFQUFFLElBQUksQ0FBQyxTQUFTO1NBQzVCLENBQUE7SUFDSCxDQUFDO0lBRUQsS0FBSyxFQUFFO1FBQ0wsS0FBSyxFQUFFLE9BQU87UUFDZCxNQUFNLEVBQUUsT0FBTztRQUNmLFNBQVMsRUFBRSxPQUFPO1FBQ2xCLFNBQVMsRUFBRSxPQUFPO1FBQ2xCLE9BQU8sRUFBRSxPQUFPO0tBQ2pCO0lBRUQsSUFBSSxFQUFFLEdBQUcsRUFBRSxDQUFDLENBQUM7UUFDWCxNQUFNLEVBQUUsRUFBRTtLQUNYLENBQUM7SUFFRixRQUFRLEVBQUU7UUFDUixPQUFPO1lBQ0wsT0FBTztnQkFDTCxlQUFlLEVBQUUsSUFBSSxDQUFDLEtBQUs7Z0JBQzNCLG1CQUFtQixFQUFFLElBQUksQ0FBQyxTQUFTO2dCQUNuQyxrQkFBa0IsRUFBRSxJQUFJLENBQUMsT0FBTztnQkFDaEMsb0JBQW9CLEVBQUUsSUFBSSxDQUFDLFNBQVM7Z0JBQ3BDLEdBQUcsSUFBSSxDQUFDLFlBQVk7YUFDckIsQ0FBQTtRQUNILENBQUM7S0FDRjtJQUVELE9BQU8sRUFBRTtRQUNQLFFBQVEsQ0FBRSxHQUFHLEVBQUUsRUFBRTtZQUNmLElBQUksQ0FBQyxNQUFNLENBQUMsSUFBSSxDQUFDLEVBQUUsR0FBRyxFQUFFLEVBQUUsRUFBRSxDQUFDLENBQUE7UUFDL0IsQ0FBQztRQUNELFVBQVUsQ0FBRSxHQUFHO1lBQ2IsTUFBTSxLQUFLLEdBQUcsSUFBSSxDQUFDLE1BQU0sQ0FBQyxTQUFTLENBQUMsQ0FBQyxDQUFDLEVBQUUsQ0FBQyxDQUFDLENBQUMsR0FBRyxLQUFLLEdBQUcsQ0FBQyxDQUFBO1lBRXZELElBQUksS0FBSyxHQUFHLENBQUMsQ0FBQyxFQUFFO2dCQUNkLElBQUksQ0FBQyxNQUFNLENBQUMsTUFBTSxDQUFDLEtBQUssRUFBRSxDQUFDLENBQUMsQ0FBQTthQUM3QjtRQUNILENBQUM7UUFDRCxTQUFTLENBQUUsR0FBRztZQUNaLElBQUksSUFBSSxDQUFDLE1BQU07Z0JBQUUsT0FBTTtZQUV2QixLQUFLLElBQUksQ0FBQyxHQUFHLElBQUksQ0FBQyxNQUFNLENBQUMsTUFBTSxFQUFFLENBQUMsRUFBRSxHQUFHO2dCQUNyQyxJQUFJLENBQUMsTUFBTSxDQUFDLENBQUMsQ0FBQyxDQUFDLEVBQUUsQ0FBQyxHQUFHLENBQUMsQ0FBQTthQUN2QjtRQUNILENBQUM7S0FDRjtJQUVELE1BQU0sQ0FBRSxDQUFDO1FBQ1AsTUFBTSxJQUFJLEdBQUc7WUFDWCxXQUFXLEVBQUUsUUFBUTtZQUNyQixPQUFPLEVBQUUsSUFBSSxDQUFDLE9BQU87U0FDdEIsQ0FBQTtRQUVELE9BQU8sQ0FBQyxDQUFDLEtBQUssRUFBRSxJQUFJLEVBQUUsQ0FBQyxJQUFJLENBQUMsTUFBTSxDQUFDLE9BQU8sQ0FBQyxDQUFDLENBQUE7SUFDOUMsQ0FBQztDQUNGLENBQUEiLCJzb3VyY2VzQ29udGVudCI6WyIvLyBTdHlsZXNcbmltcG9ydCAnLi4vLi4vc3R5bHVzL2NvbXBvbmVudHMvX2xpc3RzLnN0eWwnXG5cbi8vIE1peGluc1xuaW1wb3J0IFRoZW1lYWJsZSBmcm9tICcuLi8uLi9taXhpbnMvdGhlbWVhYmxlJ1xuaW1wb3J0IHtcbiAgcHJvdmlkZSBhcyBSZWdpc3RyYWJsZVByb3ZpZGVcbn0gZnJvbSAnLi4vLi4vbWl4aW5zL3JlZ2lzdHJhYmxlJ1xuXG4vKiBAdnVlL2NvbXBvbmVudCAqL1xuZXhwb3J0IGRlZmF1bHQge1xuICBuYW1lOiAndi1saXN0JyxcblxuICBtaXhpbnM6IFtcbiAgICBSZWdpc3RyYWJsZVByb3ZpZGUoJ2xpc3QnKSxcbiAgICBUaGVtZWFibGVcbiAgXSxcblxuICBwcm92aWRlICgpIHtcbiAgICByZXR1cm4ge1xuICAgICAgJ2xpc3RDbGljayc6IHRoaXMubGlzdENsaWNrXG4gICAgfVxuICB9LFxuXG4gIHByb3BzOiB7XG4gICAgZGVuc2U6IEJvb2xlYW4sXG4gICAgZXhwYW5kOiBCb29sZWFuLFxuICAgIHN1YmhlYWRlcjogQm9vbGVhbixcbiAgICB0aHJlZUxpbmU6IEJvb2xlYW4sXG4gICAgdHdvTGluZTogQm9vbGVhblxuICB9LFxuXG4gIGRhdGE6ICgpID0+ICh7XG4gICAgZ3JvdXBzOiBbXVxuICB9KSxcblxuICBjb21wdXRlZDoge1xuICAgIGNsYXNzZXMgKCkge1xuICAgICAgcmV0dXJuIHtcbiAgICAgICAgJ3YtbGlzdC0tZGVuc2UnOiB0aGlzLmRlbnNlLFxuICAgICAgICAndi1saXN0LS1zdWJoZWFkZXInOiB0aGlzLnN1YmhlYWRlcixcbiAgICAgICAgJ3YtbGlzdC0tdHdvLWxpbmUnOiB0aGlzLnR3b0xpbmUsXG4gICAgICAgICd2LWxpc3QtLXRocmVlLWxpbmUnOiB0aGlzLnRocmVlTGluZSxcbiAgICAgICAgLi4udGhpcy50aGVtZUNsYXNzZXNcbiAgICAgIH1cbiAgICB9XG4gIH0sXG5cbiAgbWV0aG9kczoge1xuICAgIHJlZ2lzdGVyICh1aWQsIGNiKSB7XG4gICAgICB0aGlzLmdyb3Vwcy5wdXNoKHsgdWlkLCBjYiB9KVxuICAgIH0sXG4gICAgdW5yZWdpc3RlciAodWlkKSB7XG4gICAgICBjb25zdCBpbmRleCA9IHRoaXMuZ3JvdXBzLmZpbmRJbmRleChnID0+IGcudWlkID09PSB1aWQpXG5cbiAgICAgIGlmIChpbmRleCA+IC0xKSB7XG4gICAgICAgIHRoaXMuZ3JvdXBzLnNwbGljZShpbmRleCwgMSlcbiAgICAgIH1cbiAgICB9LFxuICAgIGxpc3RDbGljayAodWlkKSB7XG4gICAgICBpZiAodGhpcy5leHBhbmQpIHJldHVyblxuXG4gICAgICBmb3IgKGxldCBpID0gdGhpcy5ncm91cHMubGVuZ3RoOyBpLS07KSB7XG4gICAgICAgIHRoaXMuZ3JvdXBzW2ldLmNiKHVpZClcbiAgICAgIH1cbiAgICB9XG4gIH0sXG5cbiAgcmVuZGVyIChoKSB7XG4gICAgY29uc3QgZGF0YSA9IHtcbiAgICAgIHN0YXRpY0NsYXNzOiAndi1saXN0JyxcbiAgICAgICdjbGFzcyc6IHRoaXMuY2xhc3Nlc1xuICAgIH1cblxuICAgIHJldHVybiBoKCdkaXYnLCBkYXRhLCBbdGhpcy4kc2xvdHMuZGVmYXVsdF0pXG4gIH1cbn1cbiJdfQ==