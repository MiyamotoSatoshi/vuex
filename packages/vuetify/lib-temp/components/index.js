export * from './VApp';
export * from './VAlert';
export * from './VAutocomplete';
export * from './VAvatar';
export * from './VBadge';
export * from './VBottomNav';
export * from './VBottomSheet';
export * from './VBreadcrumbs';
export * from './VBtn';
export * from './VBtnToggle';
export * from './VCard';
export * from './VCarousel';
export * from './VCheckbox';
export * from './VChip';
export * from './VCombobox';
export * from './VCounter';
export * from './VDataIterator';
export * from './VDataTable';
export * from './VDatePicker';
export * from './VDialog';
export * from './VDivider';
export * from './VExpansionPanel';
export * from './VFooter';
export * from './VForm';
export * from './VGrid';
export * from './VHover';
export * from './VIcon';
export * from './VImg';
export * from './VInput';
export * from './VItemGroup';
export * from './VJumbotron';
export * from './VLabel';
export * from './VList';
export * from './VMenu';
export * from './VMessages';
export * from './VNavigationDrawer';
export * from './VOverflowBtn';
export * from './VPagination';
export * from './VPaper';
export * from './VParallax';
export * from './VPicker';
export * from './VProgressCircular';
export * from './VProgressLinear';
export * from './VRadioGroup';
export * from './VRangeSlider';
export * from './VRating';
export * from './VResponsive';
export * from './VSelect';
export * from './VSlider';
export * from './VSnackbar';
export * from './VSpeedDial';
export * from './VStepper';
export * from './VSubheader';
export * from './VSwitch';
export * from './VSystemBar';
export * from './VTabs';
export * from './VTextarea';
export * from './VTextField';
export * from './VTimeline';
export * from './VTimePicker';
export * from './VToolbar';
export * from './VTooltip';
export * from './VTreeview';
export * from './VWindow';
export * from './transitions';
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi9zcmMvY29tcG9uZW50cy9pbmRleC50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQSxjQUFjLFFBQVEsQ0FBQTtBQUN0QixjQUFjLFVBQVUsQ0FBQTtBQUN4QixjQUFjLGlCQUFpQixDQUFBO0FBQy9CLGNBQWMsV0FBVyxDQUFBO0FBQ3pCLGNBQWMsVUFBVSxDQUFBO0FBQ3hCLGNBQWMsY0FBYyxDQUFBO0FBQzVCLGNBQWMsZ0JBQWdCLENBQUE7QUFDOUIsY0FBYyxnQkFBZ0IsQ0FBQTtBQUM5QixjQUFjLFFBQVEsQ0FBQTtBQUN0QixjQUFjLGNBQWMsQ0FBQTtBQUM1QixjQUFjLFNBQVMsQ0FBQTtBQUN2QixjQUFjLGFBQWEsQ0FBQTtBQUMzQixjQUFjLGFBQWEsQ0FBQTtBQUMzQixjQUFjLFNBQVMsQ0FBQTtBQUN2QixjQUFjLGFBQWEsQ0FBQTtBQUMzQixjQUFjLFlBQVksQ0FBQTtBQUMxQixjQUFjLGlCQUFpQixDQUFBO0FBQy9CLGNBQWMsY0FBYyxDQUFBO0FBQzVCLGNBQWMsZUFBZSxDQUFBO0FBQzdCLGNBQWMsV0FBVyxDQUFBO0FBQ3pCLGNBQWMsWUFBWSxDQUFBO0FBQzFCLGNBQWMsbUJBQW1CLENBQUE7QUFDakMsY0FBYyxXQUFXLENBQUE7QUFDekIsY0FBYyxTQUFTLENBQUE7QUFDdkIsY0FBYyxTQUFTLENBQUE7QUFDdkIsY0FBYyxVQUFVLENBQUE7QUFDeEIsY0FBYyxTQUFTLENBQUE7QUFDdkIsY0FBYyxRQUFRLENBQUE7QUFDdEIsY0FBYyxVQUFVLENBQUE7QUFDeEIsY0FBYyxjQUFjLENBQUE7QUFDNUIsY0FBYyxjQUFjLENBQUE7QUFDNUIsY0FBYyxVQUFVLENBQUE7QUFDeEIsY0FBYyxTQUFTLENBQUE7QUFDdkIsY0FBYyxTQUFTLENBQUE7QUFDdkIsY0FBYyxhQUFhLENBQUE7QUFDM0IsY0FBYyxxQkFBcUIsQ0FBQTtBQUNuQyxjQUFjLGdCQUFnQixDQUFBO0FBQzlCLGNBQWMsZUFBZSxDQUFBO0FBQzdCLGNBQWMsVUFBVSxDQUFBO0FBQ3hCLGNBQWMsYUFBYSxDQUFBO0FBQzNCLGNBQWMsV0FBVyxDQUFBO0FBQ3pCLGNBQWMscUJBQXFCLENBQUE7QUFDbkMsY0FBYyxtQkFBbUIsQ0FBQTtBQUNqQyxjQUFjLGVBQWUsQ0FBQTtBQUM3QixjQUFjLGdCQUFnQixDQUFBO0FBQzlCLGNBQWMsV0FBVyxDQUFBO0FBQ3pCLGNBQWMsZUFBZSxDQUFBO0FBQzdCLGNBQWMsV0FBVyxDQUFBO0FBQ3pCLGNBQWMsV0FBVyxDQUFBO0FBQ3pCLGNBQWMsYUFBYSxDQUFBO0FBQzNCLGNBQWMsY0FBYyxDQUFBO0FBQzVCLGNBQWMsWUFBWSxDQUFBO0FBQzFCLGNBQWMsY0FBYyxDQUFBO0FBQzVCLGNBQWMsV0FBVyxDQUFBO0FBQ3pCLGNBQWMsY0FBYyxDQUFBO0FBQzVCLGNBQWMsU0FBUyxDQUFBO0FBQ3ZCLGNBQWMsYUFBYSxDQUFBO0FBQzNCLGNBQWMsY0FBYyxDQUFBO0FBQzVCLGNBQWMsYUFBYSxDQUFBO0FBQzNCLGNBQWMsZUFBZSxDQUFBO0FBQzdCLGNBQWMsWUFBWSxDQUFBO0FBQzFCLGNBQWMsWUFBWSxDQUFBO0FBQzFCLGNBQWMsYUFBYSxDQUFBO0FBQzNCLGNBQWMsV0FBVyxDQUFBO0FBQ3pCLGNBQWMsZUFBZSxDQUFBIiwic291cmNlc0NvbnRlbnQiOlsiZXhwb3J0ICogZnJvbSAnLi9WQXBwJ1xuZXhwb3J0ICogZnJvbSAnLi9WQWxlcnQnXG5leHBvcnQgKiBmcm9tICcuL1ZBdXRvY29tcGxldGUnXG5leHBvcnQgKiBmcm9tICcuL1ZBdmF0YXInXG5leHBvcnQgKiBmcm9tICcuL1ZCYWRnZSdcbmV4cG9ydCAqIGZyb20gJy4vVkJvdHRvbU5hdidcbmV4cG9ydCAqIGZyb20gJy4vVkJvdHRvbVNoZWV0J1xuZXhwb3J0ICogZnJvbSAnLi9WQnJlYWRjcnVtYnMnXG5leHBvcnQgKiBmcm9tICcuL1ZCdG4nXG5leHBvcnQgKiBmcm9tICcuL1ZCdG5Ub2dnbGUnXG5leHBvcnQgKiBmcm9tICcuL1ZDYXJkJ1xuZXhwb3J0ICogZnJvbSAnLi9WQ2Fyb3VzZWwnXG5leHBvcnQgKiBmcm9tICcuL1ZDaGVja2JveCdcbmV4cG9ydCAqIGZyb20gJy4vVkNoaXAnXG5leHBvcnQgKiBmcm9tICcuL1ZDb21ib2JveCdcbmV4cG9ydCAqIGZyb20gJy4vVkNvdW50ZXInXG5leHBvcnQgKiBmcm9tICcuL1ZEYXRhSXRlcmF0b3InXG5leHBvcnQgKiBmcm9tICcuL1ZEYXRhVGFibGUnXG5leHBvcnQgKiBmcm9tICcuL1ZEYXRlUGlja2VyJ1xuZXhwb3J0ICogZnJvbSAnLi9WRGlhbG9nJ1xuZXhwb3J0ICogZnJvbSAnLi9WRGl2aWRlcidcbmV4cG9ydCAqIGZyb20gJy4vVkV4cGFuc2lvblBhbmVsJ1xuZXhwb3J0ICogZnJvbSAnLi9WRm9vdGVyJ1xuZXhwb3J0ICogZnJvbSAnLi9WRm9ybSdcbmV4cG9ydCAqIGZyb20gJy4vVkdyaWQnXG5leHBvcnQgKiBmcm9tICcuL1ZIb3ZlcidcbmV4cG9ydCAqIGZyb20gJy4vVkljb24nXG5leHBvcnQgKiBmcm9tICcuL1ZJbWcnXG5leHBvcnQgKiBmcm9tICcuL1ZJbnB1dCdcbmV4cG9ydCAqIGZyb20gJy4vVkl0ZW1Hcm91cCdcbmV4cG9ydCAqIGZyb20gJy4vVkp1bWJvdHJvbidcbmV4cG9ydCAqIGZyb20gJy4vVkxhYmVsJ1xuZXhwb3J0ICogZnJvbSAnLi9WTGlzdCdcbmV4cG9ydCAqIGZyb20gJy4vVk1lbnUnXG5leHBvcnQgKiBmcm9tICcuL1ZNZXNzYWdlcydcbmV4cG9ydCAqIGZyb20gJy4vVk5hdmlnYXRpb25EcmF3ZXInXG5leHBvcnQgKiBmcm9tICcuL1ZPdmVyZmxvd0J0bidcbmV4cG9ydCAqIGZyb20gJy4vVlBhZ2luYXRpb24nXG5leHBvcnQgKiBmcm9tICcuL1ZQYXBlcidcbmV4cG9ydCAqIGZyb20gJy4vVlBhcmFsbGF4J1xuZXhwb3J0ICogZnJvbSAnLi9WUGlja2VyJ1xuZXhwb3J0ICogZnJvbSAnLi9WUHJvZ3Jlc3NDaXJjdWxhcidcbmV4cG9ydCAqIGZyb20gJy4vVlByb2dyZXNzTGluZWFyJ1xuZXhwb3J0ICogZnJvbSAnLi9WUmFkaW9Hcm91cCdcbmV4cG9ydCAqIGZyb20gJy4vVlJhbmdlU2xpZGVyJ1xuZXhwb3J0ICogZnJvbSAnLi9WUmF0aW5nJ1xuZXhwb3J0ICogZnJvbSAnLi9WUmVzcG9uc2l2ZSdcbmV4cG9ydCAqIGZyb20gJy4vVlNlbGVjdCdcbmV4cG9ydCAqIGZyb20gJy4vVlNsaWRlcidcbmV4cG9ydCAqIGZyb20gJy4vVlNuYWNrYmFyJ1xuZXhwb3J0ICogZnJvbSAnLi9WU3BlZWREaWFsJ1xuZXhwb3J0ICogZnJvbSAnLi9WU3RlcHBlcidcbmV4cG9ydCAqIGZyb20gJy4vVlN1YmhlYWRlcidcbmV4cG9ydCAqIGZyb20gJy4vVlN3aXRjaCdcbmV4cG9ydCAqIGZyb20gJy4vVlN5c3RlbUJhcidcbmV4cG9ydCAqIGZyb20gJy4vVlRhYnMnXG5leHBvcnQgKiBmcm9tICcuL1ZUZXh0YXJlYSdcbmV4cG9ydCAqIGZyb20gJy4vVlRleHRGaWVsZCdcbmV4cG9ydCAqIGZyb20gJy4vVlRpbWVsaW5lJ1xuZXhwb3J0ICogZnJvbSAnLi9WVGltZVBpY2tlcidcbmV4cG9ydCAqIGZyb20gJy4vVlRvb2xiYXInXG5leHBvcnQgKiBmcm9tICcuL1ZUb29sdGlwJ1xuZXhwb3J0ICogZnJvbSAnLi9WVHJlZXZpZXcnXG5leHBvcnQgKiBmcm9tICcuL1ZXaW5kb3cnXG5leHBvcnQgKiBmcm9tICcuL3RyYW5zaXRpb25zJ1xuIl19