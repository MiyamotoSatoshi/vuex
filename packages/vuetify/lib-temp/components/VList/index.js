import { createSimpleFunctional } from '../../util/helpers';
import VList from './VList';
import VListGroup from './VListGroup';
import VListTile from './VListTile';
import VListTileAction from './VListTileAction';
import VListTileAvatar from './VListTileAvatar';
export { VList, VListGroup, VListTile, VListTileAction, VListTileAvatar };
export const VListTileActionText = createSimpleFunctional('v-list__tile__action-text', 'span');
export const VListTileContent = createSimpleFunctional('v-list__tile__content', 'div');
export const VListTileTitle = createSimpleFunctional('v-list__tile__title', 'div');
export const VListTileSubTitle = createSimpleFunctional('v-list__tile__sub-title', 'div');
export default {
    $_vuetify_subcomponents: {
        VList,
        VListGroup,
        VListTile,
        VListTileAction,
        VListTileActionText,
        VListTileAvatar,
        VListTileContent,
        VListTileSubTitle,
        VListTileTitle
    }
};
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi9zcmMvY29tcG9uZW50cy9WTGlzdC9pbmRleC5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQSxPQUFPLEVBQUUsc0JBQXNCLEVBQUUsTUFBTSxvQkFBb0IsQ0FBQTtBQUUzRCxPQUFPLEtBQUssTUFBTSxTQUFTLENBQUE7QUFDM0IsT0FBTyxVQUFVLE1BQU0sY0FBYyxDQUFBO0FBQ3JDLE9BQU8sU0FBUyxNQUFNLGFBQWEsQ0FBQTtBQUNuQyxPQUFPLGVBQWUsTUFBTSxtQkFBbUIsQ0FBQTtBQUMvQyxPQUFPLGVBQWUsTUFBTSxtQkFBbUIsQ0FBQTtBQUUvQyxPQUFPLEVBQUUsS0FBSyxFQUFFLFVBQVUsRUFBRSxTQUFTLEVBQUUsZUFBZSxFQUFFLGVBQWUsRUFBRSxDQUFBO0FBQ3pFLE1BQU0sQ0FBQyxNQUFNLG1CQUFtQixHQUFHLHNCQUFzQixDQUFDLDJCQUEyQixFQUFFLE1BQU0sQ0FBQyxDQUFBO0FBQzlGLE1BQU0sQ0FBQyxNQUFNLGdCQUFnQixHQUFHLHNCQUFzQixDQUFDLHVCQUF1QixFQUFFLEtBQUssQ0FBQyxDQUFBO0FBQ3RGLE1BQU0sQ0FBQyxNQUFNLGNBQWMsR0FBRyxzQkFBc0IsQ0FBQyxxQkFBcUIsRUFBRSxLQUFLLENBQUMsQ0FBQTtBQUNsRixNQUFNLENBQUMsTUFBTSxpQkFBaUIsR0FBRyxzQkFBc0IsQ0FBQyx5QkFBeUIsRUFBRSxLQUFLLENBQUMsQ0FBQTtBQUV6RixlQUFlO0lBQ2IsdUJBQXVCLEVBQUU7UUFDdkIsS0FBSztRQUNMLFVBQVU7UUFDVixTQUFTO1FBQ1QsZUFBZTtRQUNmLG1CQUFtQjtRQUNuQixlQUFlO1FBQ2YsZ0JBQWdCO1FBQ2hCLGlCQUFpQjtRQUNqQixjQUFjO0tBQ2Y7Q0FDRixDQUFBIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IHsgY3JlYXRlU2ltcGxlRnVuY3Rpb25hbCB9IGZyb20gJy4uLy4uL3V0aWwvaGVscGVycydcblxuaW1wb3J0IFZMaXN0IGZyb20gJy4vVkxpc3QnXG5pbXBvcnQgVkxpc3RHcm91cCBmcm9tICcuL1ZMaXN0R3JvdXAnXG5pbXBvcnQgVkxpc3RUaWxlIGZyb20gJy4vVkxpc3RUaWxlJ1xuaW1wb3J0IFZMaXN0VGlsZUFjdGlvbiBmcm9tICcuL1ZMaXN0VGlsZUFjdGlvbidcbmltcG9ydCBWTGlzdFRpbGVBdmF0YXIgZnJvbSAnLi9WTGlzdFRpbGVBdmF0YXInXG5cbmV4cG9ydCB7IFZMaXN0LCBWTGlzdEdyb3VwLCBWTGlzdFRpbGUsIFZMaXN0VGlsZUFjdGlvbiwgVkxpc3RUaWxlQXZhdGFyIH1cbmV4cG9ydCBjb25zdCBWTGlzdFRpbGVBY3Rpb25UZXh0ID0gY3JlYXRlU2ltcGxlRnVuY3Rpb25hbCgndi1saXN0X190aWxlX19hY3Rpb24tdGV4dCcsICdzcGFuJylcbmV4cG9ydCBjb25zdCBWTGlzdFRpbGVDb250ZW50ID0gY3JlYXRlU2ltcGxlRnVuY3Rpb25hbCgndi1saXN0X190aWxlX19jb250ZW50JywgJ2RpdicpXG5leHBvcnQgY29uc3QgVkxpc3RUaWxlVGl0bGUgPSBjcmVhdGVTaW1wbGVGdW5jdGlvbmFsKCd2LWxpc3RfX3RpbGVfX3RpdGxlJywgJ2RpdicpXG5leHBvcnQgY29uc3QgVkxpc3RUaWxlU3ViVGl0bGUgPSBjcmVhdGVTaW1wbGVGdW5jdGlvbmFsKCd2LWxpc3RfX3RpbGVfX3N1Yi10aXRsZScsICdkaXYnKVxuXG5leHBvcnQgZGVmYXVsdCB7XG4gICRfdnVldGlmeV9zdWJjb21wb25lbnRzOiB7XG4gICAgVkxpc3QsXG4gICAgVkxpc3RHcm91cCxcbiAgICBWTGlzdFRpbGUsXG4gICAgVkxpc3RUaWxlQWN0aW9uLFxuICAgIFZMaXN0VGlsZUFjdGlvblRleHQsXG4gICAgVkxpc3RUaWxlQXZhdGFyLFxuICAgIFZMaXN0VGlsZUNvbnRlbnQsXG4gICAgVkxpc3RUaWxlU3ViVGl0bGUsXG4gICAgVkxpc3RUaWxlVGl0bGVcbiAgfVxufVxuIl19