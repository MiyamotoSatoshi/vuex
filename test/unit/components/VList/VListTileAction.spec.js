import Vue from 'vue'
import { VListTileAction } from '@/components/VList'
import { test } from '@/test'

test('VListTileAction.js', ({ mount, functionalContext }) => {
  it('should render component and match snapshot', () => {
    const wrapper = mount(VListTileAction, functionalContext())

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with static class and match snapshot', () => {
    const wrapper = mount(VListTileAction, functionalContext({
      staticClass: 'static-class'
    }))

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with many children and match snapshot', () => {
    const content1 = mount(Vue.component('content1', {
      render: h => h('div')
    })).vNode
    const content2 = mount(Vue.component('content2', {
      render: h => h('span')
    })).vNode
    const wrapper = mount(VListTileAction, functionalContext({}, [content1, content2]))

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with one children and match snapshot', () => {

    const visible = mount(Vue.component('visible', {
      render: h => { if(true) return h('div') }
    })).vNode
    const notVisible = mount(Vue.component('notVisible', {
      render: h => { if(false) return h('span') }
    })).vNode

    const wrapper = mount(VListTileAction, functionalContext({}, [visible, notVisible]))

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should work with v-html', () => {
    const wrapper = mount(VListTileAction, {
      context: Object.assign({
        domProps: {
          innerHTML: '<b>something</b>'
        },
        data: {},
        props: {}
      })
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
