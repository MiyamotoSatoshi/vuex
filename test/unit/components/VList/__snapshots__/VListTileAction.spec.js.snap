// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VListTileAction.js should render component and match snapshot 1`] = `

<div class="list__tile__action">
</div>

`;

exports[`VListTileAction.js should render component with many children and match snapshot 1`] = `

<div class="list__tile__action list__tile__action--stack">
  <div>
  </div>
  <span>
  </span>
</div>

`;

exports[`VListTileAction.js should render component with static class and match snapshot 1`] = `

<div class="list__tile__action static-class">
</div>

`;
